### ISSUE TITLE
Filter call method throws error when 'items' is null

### ISSUE THREAD

**Issue by sambbaron** [angular/angular.dart#1749]
When I load a component using 'filter' on 'ng-repeat', the 'call' method is being run before the 'items' parameter is being set. So I am getting repeating errors from line 200. After a certain number of tries, 'items' gets set and the 'call' method runs without error.

    The null object does not have a method 'where'.
    NoSuchMethodError: method not found: 'where'
    Receiver: null
    Arguments: [Closure: (dynamic) => dynamic]
    
    STACKTRACE:
    #0      Object._noSuchMethod (dart:core-patch/object_patch.dart:42)
    #1      Object.noSuchMethod (dart:core-patch/object_patch.dart:45)
    #2      Filter.call (package:angular/formatter/filter.dart:200:26) 

This is happening in more than one component.  Here's an example of my template and dart code:

    <tr ng-repeat="appSession in appSessions | orderBy: orderByList | filter: filterObject">

	@NgOneWay("app-sessions") List<AppSessionModel> appSessions = [];
	@NgOneWay("mobile-user-id") int mobileUserID;
	
	void attach() {
	
	    if (appSessions == null) {
	        if (mobileUserID != null) {
	            appSessionManager.getForUser(mobileUserID).then((sessions) {
	                appSessions = sessions;
	            });
	        } else if (!fromUserTable) {
	            appSessionManager.getOpen().then((sessions) {
	                appSessions = sessions;
	            });
	        }
	    }
	}

Here, I am setting the template's list either through the attribute or attach method.  In either case, I still get the filter error.

I am not sure whether I am setting up my components incorrectly or whether there is a legitimate bug that would be solved by simply null-checking 'items' in Filter.call.

Thanks,
Sam

**Comment by zoechi** [comment]
See also http://stackoverflow.com/questions/34800537/angulardart-filter-call-method-throws-error-when-items-is-null

**Comment by sambbaron** [comment]
Thank you @zoechi.  I'll reiterate the solution I just posted to SO, which changes the underlying issue.  The error is avoided if I switch 'orderBy' and 'filter' in the template so that 'filter' is before 'orderBy'.

    <tr ng-repeat="appSession in appSessions | filter: filterObject | orderBy: orderByList">

However, I was not able to find the actual bug.  I debugged with 'orderBy' before 'filter', and OrderBy.call is returning 'items' as a list, but Filter.call is receiving 'items' as null.  I don't know what's happening between those two methods that would change 'items'.
